### PR TITLE
SDIT-643 allow retry of failed prison events 

### PR DESCRIPTION
**Pull request by steverendell**
Correct use of Oracle Connection factories

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/config/JMSConfiguration.kt [hunk 1]
```diff
@@ -1,40 +1,57 @@
...
+import org.springframework.util.ErrorHandler
 import uk.gov.justice.digital.hmpps.prisonerevents.service.JMSReceiver
 import javax.jms.ConnectionFactory
+import javax.jms.ExceptionListener
 import javax.sql.DataSource
 
-@Configuration
-class JMSConfiguration(
-  @Value("\${source.queue.name}") val queueName: String,
-  @Value("\${source.queue.transacted}") val transacted: Boolean,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/config/JMSConfiguration.kt:16]: no need for this configuration, it was only there to pave the way for integration tests using localstack

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -40,13 +40,6 @@ spring:
     hibernate:
       ddl-auto: none
 
-  datasource:
-    hikari:
-      pool-name: Nomis-Prisoner-CP
-      connectionTimeout: 1000
-      validationTimeout: 500
-      maximum-pool-size: 10
-
```
Comment (steverendell) [inline src/main/resources/application.yml:49]: not using a pool

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt [hunk 3]
```diff
@@ -66,77 +68,110 @@ class OracleToTopicIntTest : IntegrationTestBase() {
...
+    @Test
+    fun `Retry exception messages when messages present`() {
+      sabotageTopic()
+      repeat(2) { simulateTrigger() }
+      await.atMost(Duration.ofSeconds(20)) untilCallTo { getGetNumberOfMessagesCurrentlyOnExceptionQueue() } matches { it == 2 }
+      fixTopic()
+
+      webTestClient
+        .mutate()
+        .responseTimeout(Duration.ofSeconds(2000))
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerevents/integration/OracleToTopicIntTest.kt:165]: will remove this

**Pull request merged**